relayer/chains/evm/listener.go

```diff
@@ -106,13 +110,14 @@ func (p *Provider) Listener(ctx context.Context, lastSavedHeight uint64, blockIn
 							Addresses: p.blockReq.Addresses,
 							Topics:    p.blockReq.Topics,
 						}
-						p.log.Info("syncing", zap.Uint64("start", br.start), zap.Uint64("end", br.end), zap.Uint64("latest", latestHeight))
+						p.log.Info("syncing", zap.Uint64("start", br.start), zap.Uint64("end", br.end), zap.Uint64("latest", latestHeight), zap.Uint64("delta", latestHeight-br.end))
 						logs, err := p.getLogsRetry(ctx, filter, br.retry)
 						if err != nil {
 							p.log.Warn("failed to fetch blocks", zap.Uint64("from", br.start), zap.Uint64("to", br.end), zap.Error(err))
 							continue
 						}
-						p.log.Info("synced", zap.Uint64("start", br.start), zap.Uint64("end", br.end), zap.Uint64("latest", latestHeight))
+						p.log.Info("synced", zap.Uint64("start", br.start), zap.Uint64("end", br.end), zap.Uint64("latest", latestHeight), zap.Uint64("delta", latestHeight-br.end))
+						time.Sleep(BlockSyncInterval)
```

Comment:
What is the reason to sleep this goroutine for block sync interval?

Reply:
3s might be a bit too much if it is only to avoid the rate limit. Going from unlimited to ~0.33req/second here. Was this number selected via some logic or just randomly ?

Reply:
With some consideration yeah.

This number is chosen taking other actions into account such as active subscriptions, ongoing transactions, other background RPC calls, etc.

Because this happens so infrequently and we have massive block range we could do and concurrency hurts us than the benefit.

Reply:
This is only when syncing right? So 3 seconds per batch of blocks? 

Usually many seem to have Per second limits of request so pretty much any type of sleep should suffice to circumvent that. But if batches are big enough syncing with 3 seconds per batch should not be to bad unless we are very far behind

Reply:
Yep. mainly used for recovering only when websocket fails.

Reply:
But we may have other calls when syncing so we need to be careful.

Reply:
hmm yeah, it's a bit hacky to add sleep timers but if we have no more sophisticated way of rate limiting ourselves.  If we ever build some kind of load balancer/RPC manager part to the relayer we could rate have rate limits in here to pace the relayer in one place

Reply:
We also have logic to sync the latest height as last processed height which runs every 5 minutes and updates the last processed block, so adding 3s sleep here might cause some problems when the sync time would take more than 5 minutes (for around more than 1000 blocks). Shouldn't it be verified as well if it will have any effects ?